### ISSUE TITLE
Config cache error: `dependencies` cannot serialize object of type 'org.gradle.api.artifacts.Configuration'

### ISSUE THREAD

**Issue by yogurtearl** [gradle/gradle#17471]
Run `./gradlew dependencies --configuration-cache`

### Expected Behavior

No config cache errors

### Current Behavior

Got this error: 
```
- Task `:dependencies` of type `org.gradle.api.tasks.diagnostics.DependencyReportTask`: cannot serialize object of type 'org.gradle.api.artifacts.Configuration' as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.1/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
```

```
⌄⚠️cannot serialize object of type org.gradle.api.artifacts.Configuration📋 as these are not supported with the configuration cache. ?
⌄fieldconfiguration📋 of org.gradle.api.tasks.diagnostics.internal.dependencies.AsciiDependencyReportRenderer$ConfigurationAction📋
⌄bean of type org.gradle.api.tasks.diagnostics.internal.dependencies.AsciiDependencyReportRenderer$ConfigurationAction📋
⌄fieldconfigurationAction📋 of org.gradle.api.tasks.diagnostics.internal.dependencies.AsciiDependencyReportRenderer📋
⌄bean of type org.gradle.api.tasks.diagnostics.internal.dependencies.AsciiDependencyReportRenderer📋
⌄fieldrenderer📋 of org.gradle.api.tasks.diagnostics.DependencyReportTask📋
■task:dependencies📋 of type org.gradle.api.tasks.diagnostics.DependencyReportTask📋
```

### Your Environment

Gradle 7.1

**Comment by mlopatkin** [comment]
@gradle/configuration-cache is it a duplicate of #12871?

**Comment by mlopatkin** [comment]
Duplicate of #12871

**Comment by jhansche** [comment]
Should this be reopened?

#12871 was resolved as of Gradle 7.5, but in 7.5.1 the DependencyReportTask still does not use the new APIs, and still reports configuration cache problems, due to use of getProject() and trying to serialize Configuration.

So now this issue seems to be relevant again, because the dependencies report now has to be updated to use the new APIs.

**Comment by Mary-a67** [comment]
This issue happens with Gradle 7.5.1, should this be opened or create a new bug report?
this is a blocker to upgrading Gradle enterprise V 3.11.4
@mlopatkin

**Comment by mlopatkin** [comment]
> This issue happens with Gradle 7.5.1, should this be opened or create a new bug report? this is a blocker to upgrading Gradle enterprise V 3.11.4 @mlopatkin

Apparently, I chose the duplicated issue poorly. The correct one is #17470, and it was fixed only in 7.6.

**Comment by mlopatkin** [comment]
Duplicate of #17470
